homeassistant/components/webostv/__init__.py

```diff
@@ -84,7 +87,49 @@
 
 async def async_setup(hass, config):
     """Set up the LG WebOS TV platform."""
-    hass.data[DOMAIN] = {}
+    hass.data.setdefault(DOMAIN, {})
+    if DOMAIN not in config:
+        return True
+
+    for conf in config[DOMAIN]:
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": SOURCE_IMPORT}, data=conf
+            )
+        )
+
+    return True
+
+
+def _async_migrate_options_from_data(hass, config_entry):
+    """Migrate options from data."""
+    if config_entry.options:
+        return
+
+    config = config_entry.data
+    options = {}
+
+    # Get Turn_on service
+    options[CONF_ON_ACTION] = config.get(CONF_ON_ACTION)
+
+    # Get Preferred Sources
+    if sources := config.get(CONF_CUSTOMIZE, {}).get(CONF_SOURCES):
+        options[CONF_SOURCES] = sources
+        if isinstance(sources, list) is False:
+            options[CONF_SOURCES] = sources.split(",")
+
+    hass.config_entries.async_update_entry(config_entry, options=options)
+
+
+async def async_setup_entry(hass, config_entry):
+    """Set the config entry up."""
+    _async_migrate_options_from_data(hass, config_entry)
+
+    host = config_entry.data[CONF_HOST]
+    key = config_entry.data[CONF_CLIENT_SECRET]
+
+    client = await WebOsClient.create(host, client_key=key)
+    await async_connect(client)
 
     async def async_service_handler(service):
```

Comment:
Later we should migrate these services to entity services.

Reply:
I will make some additional cleanup PRs after this one, My first intention was just to pick up this PR to finish what's left but it turned out to be bigger than I assumed.